doc/advcfg.1
.TH "AdvanceMAME Automatic Video Configurator" 1
.SH NAME
advcfg \(hy AdvanceMAME Automatic Video Configurator
.SH SYNOPSIS 
advcfg [\(hyadvmamec] [\(hyadvmessc] [\(hyadvmenuc]
.PD 0
.PP
.PD
	[\(hylog] [\(hylogsync] [\(hyrc RCFILE] [\(hybit N]
.PD 0
.PP
.PD
.SH DESCRIPTION 
The \(oqadvcfg\(cq utility is the automatic video configuration
program for AdvanceMAME, AdvanceMESS and AdvanceMENU.
.PP
This utility works differently if one of the \(hyadvmamec,
\(hyadvmessc and \(hyadvmenuc option is specified.
The main difference is the name of the configuration
file used to store the video modes.
.PP
All the video modes used in the configuration process
have always negative horizontal and vertical sync polarity.
.PP
Check the \(oqadvdev.txt\(cq file for more details on the video
drivers supported.
.SH OPTIONS 
.TP
.B \(hyrc RCFILE
Specify an alternate name of the configuration file.
.TP
.B \(hylog
A very detailed log of operations is saved in
a \(oq.log\(cq file. Very useful for debugging problems.
.TP
.B \(hylogsync
Like the \(oq\(hylog\(cq option but the log file is not
buffered. This option must be used to get a complete
log file in presence of a machine crash.
.TP
.B \(hyadvmamec, \(hyadvmessc, \(hyadvmenuc
Select the mode of operation for the programs
\(oqadvmame\(cq, \(oqadvmess\(cq and \(oqadvmenu\(cq.
The default is for \(oqadvmame\(cq.
.TP
.B \(hybit N
Select the bit depth of the test video modes.
If omitted 8 bit modes are used.
Valid values are 8, 15, 16 and 32.
.SH CONFIGURATION 
Depending one the command line options used one of the
\(oqadvmame.rc\(cq, \(oqadvmess.rc\(cq and \(oqadvmenu.rc\(cq file is used
to load and save the video configuration.
.PP
If the configuration process completes with success the
program adds these options in your configuration file:
.PP
.RS 4
device_video_clock ?
.PD 0
.PP
.PD
device_video_format ?
.PD 0
.PP
.PD
.RE
.PP
which define the capabilities of your monitor and video board.
And for the \(oqadvmame\(cq and \(oqadvmess\(cq modes also the options:
.PP
.RS 4
display_mode auto
.PD 0
.PP
.PD
display_adjust generate_yclock
.PD 0
.PP
.PD
.RE
.PP
which select the \(oqautomatic\(cq video mode generation of the
emulators.
.PP
All these options are documented in the \(oqadvdev.txt\(cq
and \(oqadvmame.txt\(cq files.
.SH COPYRIGHT 
This file is Copyright (C) 2003, 2004 Andrea Mazzoleni.


doc/history.txt
AdvanceMAME Version 0.61.0 2002/07
	) Removed the input_analog[] and input_track[] options. They
		are now substituted by the new input_map[] option.
	) Upgraded at the SVGALIB 1.9.15 library.
	) Added the `vertical' and `horizontal' sections on the
		config file. With these sections you can have
		different configuration for vertical and horizontal
		games. If you have a rotating monitor, you can now
		always configure the correct orientation.
	) The `display_rotate auto' option now selects always the
		`blit' rotation. The reason of this change is that
		the `core' rotation will be completly removed on the
		next MAME version (0.62).
		On monoprocessor machines this may result in a
		sensible slowdown on some games. Instead, on SMP
		(multiprocessor) machines all these games are now
		faster because the rotation is done in the parallel
		blit thread.
	) Fixed the joystick problem with more than 8 buttons
		[Anthony D. Saxton (las_vegas)].
	) Added the `display_intensity' option.
	) Added the `display_artwork_crop' option.
	) Fixed the `display_expand' option for the vertical vector
		games.
	) Prevented the use on the NT platform.
	) Added some new MMX blitters to speedup the blit rotation.
	) In the Linux version if the HOME environment variable
		is not set all the files are read and written on the
		PREFIX/share/advance directory.
	) Fixed a bug when forcing the use of the vbeline/3dfx driver.
	) Enabled the interlaced modes with the 3dfx svgalib driver.
	) Minor bugfix.

AdvanceMAME Version 0.60.1 2002/05
	) Fixed the MNG recording for some games.
	) Added the input_hotkey option to disable the special
		key sequences like CTRL+ALT+DEL.
	) Fixed the rgb 16pix effect on 32 bit modes.
	) Disabled the pause key. It should not be used.
	) The advs utility now prevent the keyboard buffer to be
		filled.
	) Added a new `License' chapter at the faq.txt file.

AdvanceMAME Version 0.60.0 2002/05
	) Upgraded at the SVGALIB 1.9.14 library.
	) Solved a video mode restore problem with the
		vbeline drivers.
	) After unpausing the idle exit timeout is now reset.
	) Fixed the idle_exit timing on Linux.
	) Fixed the size of vector games when a custom mode is
		specified with the display mode option.
	) Other minor bugfix.

AdvancePAC Version 0.58.0 2002/04
	) Added the PacMAME target at AdvanceMAME.

AdvanceMAME Version 0.59.1 2002/04
	) Upgraded at the SVGALIB 1.9.13 library.
	) Upgraded at the Allegro 4.01 library.
	) Fixed some issues if no sound card is present.
	) Fixed a bug in the blit rotation.
	) Updated the list of the games which require the blit
		rotation for the auto option.
	) Fixed the use of gamma and brightness on the rgb games.
	) The precision of the frame rate of the generated MNG
		files is now always at least 1:1000000.
	) The `mng' recording now can be started and stopped also
		during the pause.
	) The `script_play' option now works also if
		`misc_startuptime' is 0.
	) Fixed some issues on the directory paths.

AdvanceMAME Version 0.59.0 2002/03
	) Added the option "-r" at the zmng utility. This fixs the
		bug when compressing 24 bit image which cannot be
		converted to 8 bit.
	) Added the device_video_singlescan/doublescan/interlace
		options to disable the use of some mode types.
	) Rewritten the `vgaline' DOS driver. Now it never uses the
		BIOS to set the video mode.
	) The MNG files now use the FRAME chuck to specify a
		precise timing.
	) Added the `misc_quiet' option to disable the text message :
		"AdvanceMAME - Copyright (C) 1999-2002 by..."
	) Fixed some imprecisions on the image stretch.
	) Completed the color conversion functions.

AdvanceMAME Version 0.58.1 2002/03
	) Fixed the clock values printed on the first line of the
		video menu.
	) Changed the behaviour of the "display_adjust x" option.
		Now all the modes are adjusted in size, not only the
		nearest mode.
	) Faster `scale2x' effect. Now optimized for AGP bus.
	) Fixed the video and sound recording for the SMP machine.
	) The .mng and .wav file are now saved always with the same
		basename.
	) Fixed the selection of the video mode with the nearest clock.
	) The idleexit feature now detect also the joystick and
		mouse use.
	) MMX implementation of some functions of the tilemap engine.

AdvanceMAME Version 0.58.0 2002/02
	) Added the video recording feature in `.mng' files.
	) Added the `record_sound', `record_sound_time',
		`record_video', `record_video_time',
		`record_video_interleave' options.
	) Added the `zmng' compression utility in the contrib/zmng dir.
	) When recording, the sound is now generated ignoring
		the syncronization issues to get a perfect sound
		also with games with a very high frame skip.
	) Now you can record sound and video without any length limit.
	) The recorded sound is now NOT adjusted in volume.
	) Minor corrections at the sound syncronization.

AdvanceMESS Version 0.56.0 2002/01
	) Added the MESS target at AdvanceMAME.

AdvanceMAME Version 0.57.1 2002/01
	) The default rom and sample directory are now `rom' and
		`sample' (singular).
	) The Linux config directories are now named `$home/.advance' and
		`$prefix/share/advance'. Rename them manually.
	) Fixed the sound snapshot saving.
	) Fixed the graphics snapshot saving in Linux.
	) Fixed the dir rights in creation in Linux.
	) Fixed the support of roms not zipped.
	) Fixed the `display_resize' option saving.
	) Recompiled the video driver with lower optimizations.
	) Readded the autocentering control menu.
	) Fixed a SIGSEGV in the mode selection.
	) Other minor bugfix.

AdvanceMAME Version 0.57.0 2002/01
	) Improved the SMP performance. Now the blit stage is completly
		done by the second thread without any bitmap copy.
	) Added a Console Frame Buffer video driver for Linux.
	) Added the `misc_speed' option to control the speed of the game.
	) Added the `sound_latency' option to control the size of the
		sound buffer.
	) Added the `advs' utility. A wav/mp3 player.
	) Added the `faq.txt' and `tips.txt' files.

doc/advcfg.d
Name
	advcfg - The AdvanceCFG Automatic Configurator

Synopsis
	:advcfg [-advmamec] [-advmessc] [-advmenuc]
	:	[-log] [-logsync] [-rc RCFILE] [-bit N]

Description
	The `advcfg' utility is the automatic video configuration
	program for AdvanceMAME, AdvanceMESS and AdvanceMENU.

	This utility works differently if one of the -advmamec,
	-advmessc and -advmenuc option is specified.
	The main difference is the the name of the configuration
	file used to store the video modes.

Options
	-rc RCFILE
		Specify an alternate name of the configuration file.

	-log
		A very detailed log of operations is saved in
		a `.log' file. Very useful for debugging problems.

	-logsync
		Like the `-log' option but the log file is not
		buffered. This option must be used to get a complete
		log file in presence of a machine crash.

	-advmamec, -advmessc, -advmenuc
		Select the mode of operation for the programs `advmame',
		`advmess' and `advmenu'.
		The default is for `advmame'.

	-bit N
		Select the bit depth of the test video modes.
		If omitted the 8 bit modes are used.
		Valid values are 8, 15, 16 and 32.

Configuration
	Depending one the command line options used one of the `advmame.rc',
	`advmess.rc' and `advmenu.rc' file is used to load and save
	the video configuration.

	Check the `advdev.txt' file for more details on the video drivers
	and video options supported.

Copyright
	This file is Copyright (C) 2003 Andrea Mazzoleni.

